### PR TITLE
feat(Core/Player) Option to adjust max rest bonus via config

### LINKED ISSUE #20759: Feature: Configurable rest bones max experience
### Describe your feature request or suggestion in detail

My friends have unstable online times, and they want to accumulate more rest bonus XP while offline.

### Describe a possible solution to your feature or suggestion in detail

The code is in src/server/game/Entities/Player/Player.cpp, line 10229.
Could you make "rest_bonus_max" configurable?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Exitare**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes #20759

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Update worldserver config to include new rate option
2. Adjust the multiplier rest rate to increase the max rest bonus
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Exitare** [comment]
@Kitzunu I have questions regarding the current name & place of the option. Right now it works as a multiplier that affects the rest max rest calculation

`  // Fetch rest bonus multiplier from cached configuration
    float rest_bonus_multiplier = sWorld->getRate(RATE_REST_MAX_MULTIPLIER);

   // Calculate rest bonus max using the multiplier
   float rest_bonus_max = (float)GetUInt32Value(PLAYER_NEXT_LEVEL_XP) * rest_bonus_multiplier / 2;
  `
  
  We might need a better name for the config.

File: src/server/apps/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -2106,14 +2106,17 @@ Rate.Loyalty           = 1
...
 #        Description: Resting points grow rates.
 #        Default:     1 - (Rate.Rest.InGame)
 #                     1 - (Rate.Rest.Offline.InTavernOrCity)
 #                     1 - (Rate.Rest.Offline.InWilderness)
+#                     1 - (Rate.Rest.MaxLimit.Multiplier)
 
 Rate.Rest.InGame                 = 1
 Rate.Rest.Offline.InTavernOrCity = 1
 Rate.Rest.Offline.InWilderness   = 1
+Rate.Rest.Max.Multiplier    = 1.5
```
Comment (Kitzunu) [inline src/server/apps/worldserver/worldserver.conf.dist:2119]:
```suggestion
#                     1.5 - (Rate.Rest.MaxBonus) (Blizzlike)

Rate.Rest.InGame                 = 1
Rate.Rest.Offline.InTavernOrCity = 1
Rate.Rest.Offline.InWilderness   = 1
Rate.Rest.Max.Multiplier         = 1.5
```

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -10215,7 +10215,11 @@ void Player::SetRestBonus(float rest_bonus_new)
     if (rest_bonus_new < 0)
         rest_bonus_new = 0;
 
-    float rest_bonus_max = (float)GetUInt32Value(PLAYER_NEXT_LEVEL_XP) * 1.5f / 2;
+    // Fetch rest bonus multiplier from cached configuration
+    float rest_bonus_multiplier = sWorld->getRate(RATE_REST_MAX_MULTIPLIER);
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.cpp:10219]: rename it to restBonusMultiplier (https://www.azerothcore.org/wiki/cpp-code-standards#naming-of-variables-and-functions)

**Comment by Exitare** [comment]
@Kitzunu  thanks for your review. I addressed your requests and changed the names of both the config and the variable naming.

File: src/server/game/Entities/Player/Player.cpp [hunk 3]
```diff
@@ -10206,22 +10206,25 @@ void Player::LeaveAllArenaTeams(ObjectGuid guid)
     } while (result->NextRow());
 }
 
-void Player::SetRestBonus(float rest_bonus_new)
+void Player::SetRestBonus(float restBonusNew)
```
[the remark this answers is not in the document]
Comment (Exitare) [reply src/server/game/Entities/Player/Player.cpp:10209]: Done

**Comment by Exitare** [comment]
Does anyone have any reviews or feedback for this PR? If not, we could label it "Ready to be tested", if thats ok with everyone.

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Looks good to me

**Pull request merged**